sdk/python/kfp/compiler/compiler.py

```diff
@@ -410,6 +500,15 @@ def _create_templates(self, pipeline):
 
     new_root_group = pipeline.groups[0]
 
+    # Generate core data structures to prepare for argo yaml generation
```

Comment:
Thanks for the comments here. It's very helpful.

Reply:
Sure thing.